integration-cli/docker_cli_run_unix_test.go

```diff
@@ -768,9 +768,8 @@ func (s *DockerSuite) TestRunInvalidCPUShares(c *check.C) {
 	expected := "The minimum allowed cpu-shares is 2"
 	c.Assert(out, checker.Contains, expected)
 
-	out, _, err = dockerCmdWithError("run", "--cpu-shares", "-1", "busybox", "echo", "test")
-	c.Assert(err, check.NotNil, check.Commentf(out))
-	expected = "shares: invalid argument"
+	out, _ = dockerCmd(c, "run", "--cpu-shares", "-1", "busybox", "echo", "test")
+	expected = "WARNING: Invalid CPUShares value. Must be nonnegative. Discarding."
```

Comment:
Isn't this already validated though?

```bash
$ docker run --cpu-shares=-1 busybox echo test 
docker: Error response from daemon: linux runtime spec resources: shares: invalid argument.
ERRO[0000] error waiting for container: context canceled
```

Error message could be better, but looks like there's already code in place to handle invalid values?

Reply:
@thaJeztah Thank you for your feedback. It seems that we get an error if an inappropriate `CPUShares` value is passed. According to the strategy on invalid fields on Solaris and Unix, which output a warning on the invalid values and change these values to 0, it might be better to keep these handling strategy in the same manner. Also, it would failed earlier if we had handled this in the verification function.

Reply:
I agree an error for an invalid value is more correct than a warning.

Reply:
@dnephin I will modify these verification so that we get errors for invalid values instead warnings. BTW, some verification output warnings instead errors on the Solaris. Do we need to improve all these to keep our invalid value handling in the same manner?

Reply:
I'm not sure what the state of the solaris build is at the moment, I would leave it as-is.